pyLSV2/client.py

```diff
@@ -2213,3 +2216,144 @@ def tst_record_data(
+            self.tst_decode_signal_details(signal_list, result)
+        else:
+            if self.last_error.e_code == 85:
+                self._logger.warning("too many signals selected: %d", len(signal_list))
+                raise Exception("too many signals selected???")
+            if self.last_error.e_code == lc.LSV2StatusCode.T_ER_OSZI_CHSEL:
+                self._logger.warning("Error setting up the channels")
+                raise Exception("Error setting up the channels")
+            raise Exception()
+        
+        # for i in signal_list:
+        #     print(i)
+
+        # step 3
+        result = self._send_recive(lc.CMD.R_DT, None, lc.RSP.S_DT)
+        if isinstance(result, (bytearray,)) and len(result) > 0:
+            ts = self.decode_timestamp(result)
+            self._logger.debug("Time on Control is %s", ts.isoformat())
+        else:
+            raise Exception()
+
+        # step 4
+
+        """
+        capture 1: one channel tree axes 3k us interval
+        "\x00\x06\xff\xff\x00\x00\x00\x00\x00\x00\x00\x00 \x00\x00\x0b\xb8"
+
+
+        capture 2: two channels 1 axes 3k us interval
+        "\x00\x06\xff\xff\x00\x00\x00\x00\x00\x00\x00\x00 \x00\x00\x0b\xb8"
+
+
+        capture 3: two channels 1 axes 3k us interval 16 k buffer
+        "\x00\x06\xff\xff\x00\x00\x00\x00\x00\x00\x00\x00 \x00\x00\x0b\xb8"
+       
+        capture 4: two channels 1 axes 6k us interval
+        "\x00\x06\xff\xff\x00\x00\x00\x00\x00\x00\x00\x00 \x00\x00\x17\x70"
+
+
+        # bytes 12 - 15 are a repeat of the interval value? 3000 and 6000
+        """
+        payload = bytearray()
+        payload.append(0x00)
+        payload.append(0x06)
+        payload.append(0xFF)
+        payload.append(0xFF)
+        payload.append(0x00)
+        payload.append(0x00)
+        payload.append(0x00)
+        payload.append(0x00)
+        payload.append(0x00)
+        payload.append(0x00)
+        payload.append(0x00)
+        payload.append(0x00)
+
+        # interval again?
+        payload.extend(struct.pack("!L", intervall_us))
+        # payload.append(0x00)
+        # payload.append(0x00)
+        # payload.append(0x0b) # -> 0B
+        # payload.append(0xb8) # -> B8
+
+        recorded_data = list()
+        content = self._send_recive(lc.CMD.R_OD, payload, lc.RSP.S_OD)
+
+        if isinstance(content, (bytearray,)) and len(content) > 0:
+            recorded_data.append(self.tst_decode_scope_reading(signal_list, content))
+
+            end = time.time() ; timer = end - start ; TCP_package_num = 0
+            while timer < time_readings:
+                content = self._llcom.telegram(lc.RSP.T_OK)
+                if self._llcom.last_response in lc.RSP.S_OD:
+                    recorded_data.append(
+                        self.tst_decode_scope_reading(signal_list, content)
+                    )
+                    yield recorded_data[TCP_package_num]["signals"]
```

Comment:
I think there should be some kind of cleanup of the recoded data after it gets handed over to the application code. Currently recoded_data will increase in size until either the time_readings is reached, the application loop stops or we run out of memory.
Can you implement something so that only the data is kept here that has not been read by the application? This would make the application responsible for data storage which is preferable. Especially if the user intends to keep this running for very long times.

Reply:
I made recorded_data restart itself after each iteration.